client/src/lib/store/track.ts

```diff
@@ -29,36 +29,62 @@ interface TrackState {
   moveApplication: (
     from: ApplicationStage,
     to: ApplicationStage,
-    applicationId: string
+    applicationId: string,
+    sourceIndex: number,
+    destinationIndex: number
   ) => void;
 }
 
+// Helper function to reorder items in a list
+const reorder = (list: Application[], startIndex: number, endIndex: number) => {
+  const result = Array.from(list);
+  const [removed] = result.splice(startIndex, 1);
+  result.splice(endIndex, 0, removed);
+  return result;
+};
+
 export const useTrack = create<TrackState>()((set) => ({
   Applied: [],
   Rejected: [],
   "Online Assesstment": [],
   Interviewing: [],
   Offer: [],
+  
   addApplication: (to, application) =>
     set((state) => {
-      console.log("hellow");
       state[to].push(application);
       return { ...state };
     }),
-  removeApplication: () => {},
-  moveApplication: (from, to, applicationId) =>
+    
+  removeApplication: (from, applicationId) =>
     set((state) => {
-      const movedApplication = state[from].find(
-        (application) => application.id === applicationId
-      );
-
-      // removes application
       state[from] = state[from].filter(
         (application) => application.id !== applicationId
       );
+      return { ...state };
+    }),
 
-      // moves application to new stage
-      state[to] = [movedApplication!, ...state[to]];
+  moveApplication: (from, to, applicationId, sourceIndex, destinationIndex) =>
```

Comment:
why did the logic for the moveApplication have to be changed including the parameters?

Reply:
This allow the user to reorder the job card within the column base on the job they want to prioritize monitor.

Reply:
Got it